parser/prism/Translator.cc

```diff
@@ -792,22 +820,63 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node, bool preserveCon
 
             int numPosArgs = args.size() - (hasKwargsHash ? 1 : 0);
 
+            if (prismBlock != nullptr && PM_NODE_TYPE_P(prismBlock, PM_BLOCK_ARGUMENT_NODE) && !blockPassArgIsSymbol) {
```

Comment:
This is analogous to this Desugar logic:

https://github.com/sorbet/sorbet/blob/700540c446e5a5fa4e0cb2ba42ddf61a03f3317b/ast/desugar/Desugar.cc#L919-L935

Except we can use an early return, and disentangle it out of that deeply nested `if` pyramid.